core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -170,6 +170,21 @@ trait Endpoint[F[_], A] { self =>
       final override def toString: String = self.toString
     }
 
+  /**
+    * Transform effect of endpoint given natural transformation `F ~> G`
+    */
+  final def mapK[G[_]](nat: F ~> G): Endpoint[G, A] =
```

Comment:
This seems to be a generally adopted name for this (at least across cats). Although, I like fs2's naming schema for this a little bit better. They use `translate`, which I think fits a little nicer into our set of APIs. What do you think?

I'm not feeling super strong about this and open for merging this as is. Just decided to mention this other name that I found.

Reply:
My idea is that since we're relying on cats type classes and `FunctionK` type, it's natural to follow this convention